### PR TITLE
Adding more np to p test cases

### PR DESCRIPTION
**Pull request by simplynaveen20**
Below test scenarios will be added along with rntbd api version fix
1.Adding new item to fixed partition collection with valid SystemKeyName partitionKey
2.Adding new item to fixed partition collection with PartitionKey.None
3.Updating existing item with PartitionKey.None

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/CosmosClient.cs [hunk 1]
```diff
@@ -73,6 +74,7 @@ public class CosmosClient : IDisposable
         static CosmosClient()
         {
             HttpConstants.Versions.CurrentVersion = HttpConstants.Versions.v2018_12_31;
+            HttpConstants.Versions.CurrentVersionUTF8 = Encoding.UTF8.GetBytes(HttpConstants.Versions.v2018_12_31);
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/CosmosClient.cs]: Encoding.UTF8.GetBytes(HttpConstants.Versions.CurrentVersion);

File: Microsoft.Azure.Cosmos/src/Routing/AddressResolver.cs [hunk 2]
```diff
@@ -451,7 +451,7 @@ private void ThrowIfTargetChanged(DocumentServiceRequest request, PartitionKeyRa
                 out effectivePartitionKeyStringObject))
             {
                 // Allow EPK only for partitioned collection (excluding migrated fixed collections)
-                if (!collection.HasPartitionKey || collection.PartitionKey.IsSystemKey.GetValueOrDefault(false))
+                if (! (collection.HasPartitionKey && collection.PartitionKey.IsSystemKey.GetValueOrDefault(false)))
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/Routing/AddressResolver.cs]: Undo.

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosItemTests.cs [hunk 3]
```diff
@@ -743,18 +744,49 @@ public async Task ReadNonPartitionItemAsync()
...
                 Assert.IsNotNull(response.Resource.id == nonPartitionItemId);
 
+                //Adding item to fixed container with CosmosContainerSettings.NonePartitionKeyValue.
+                ToDoActivity itemWithoutPK = CreateRandomToDoActivity();
+                CosmosItemResponse<ToDoActivity> createResponseWithoutPk = await fixedContainer.Items.CreateItemAsync<ToDoActivity>(
+                 partitionKey: CosmosContainerSettings.NonePartitionKeyValue,
+                 item: itemWithoutPK);
+
+                Assert.IsNotNull(createResponseWithoutPk.Resource);
+                Assert.IsTrue(createResponseWithoutPk.StatusCode == HttpStatusCode.Created);
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosItemTests.cs]: Prefer Assert.AreEqual

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/ContractTests.cs [hunk 4]
```diff
@@ -24,6 +25,7 @@ public void ApiVersionTest()
             { }
 
             Assert.AreEqual(HttpConstants.Versions.v2018_12_31, HttpConstants.Versions.CurrentVersion);
+            Assert.IsTrue(Encoding.UTF8.GetBytes(HttpConstants.Versions.v2018_12_31).SequenceEqual(HttpConstants.Versions.CurrentVersionUTF8));
```
[the remark this answers is not in the document]
Comment (simplynaveen20) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/ContractTests.cs]: Byte array need to be compared with SequenceEqual , equals will compare there reference., and will fail.

[the remark this answers is not in the document]
Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/ContractTests.cs]: CollectionAssert.AreEqual()

**Pull request merged**